src/scheduler/workers/analytics/addAnalyticsAuditEvent.ts

```diff
@@ -0,0 +1,40 @@
+import { prisma } from "scheduler/workers/model";
+
+const truncateMessage = "... truncated due to max length";
+const MAX_BYTES = 7000 - Buffer.byteLength(truncateMessage);
+
+export async function addAnalyticsAuditEvent(view: string, dataSent: any[], response: any) {
+  const maxSize = MAX_BYTES - getByteLength(JSON.stringify({ data: "", response }));
+  const truncatedData = truncateString(JSON.stringify(dataSent), maxSize);
+
+  await prisma.audit.create({
+    data: {
+      auditEvent: "ANALYTICS",
+      createdAt: new Date(),
+      jsonData: {
+        view,
+        dataSent: truncatedData,
+        response,
+      },
+    },
+  });
+}
+
+function getByteLength(string: string) {
+  return Buffer.byteLength(string);
+}
+
+function truncateString(string: string, maxByteLength: number, hasTruncated = false): string {
```

Comment:
Out of curiosity, would there have been a way to split the string and reduce the length that way instead of creating a recursive function?

Reply:
`while` would be the other option. It's the same difference. In both cases we would keep having to check that we've cut off "enough". I don't anticipate it being an issue, but seeing as it's likely we have characters that would be over 2 bytes in our db (international alphabet/chars), I'm just being overly cautious incase a view/query does include it. (Otherwise I would do a cheap calculation 1 char = 2 bytes)